src/server/json_family.cc

```diff
@@ -157,6 +159,48 @@ OpResult<json> GetJson(const OpArgs& op_args, string_view key) {
   return decoder.get_result();
 }
 
+string ConvertToJsonPointer(const string& json_path) {
+  if (json_path.empty() || json_path[0] != '$') {
+    return "";
+  }
+
+  vector<string> parts;
+  bool invalid_format = false;
+  std::size_t i = 1;
+  while (i < json_path.size()) {
+    bool is_array = false;
+    bool is_object = false;
+    if (json_path[i] == '[' && json_path[i + 1] == '\'') {
+      is_object = true;
+      i += 2;
+    }
+    else if (json_path[i] == '[') {
+      is_array = true;
+      i += 1;
+    }
+
+    if (is_array) {
+      auto end_pos = json_path.find(']', i);
+      parts.emplace_back(json_path.substr(i, end_pos - i));
+      i = end_pos + 1;
+    } else if (is_object) {
+      auto end_pos = json_path.find("']", i);
```

Comment:
hmm, I am not sure it's the right way to do it. what about strings that have `']` in them?